common/persistence/nosql/shardedNosqlStore.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2022 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package nosql
+
+import (
+	"fmt"
+	"sync"
+
+	p "github.com/uber/cadence/common/persistence"
+
+	"github.com/uber/cadence/common/config"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/log/tag"
+)
+
+type (
+	// shardedNosqlStore is a store that may have one or more shards
+	shardedNosqlStore struct {
+		sync.RWMutex
+
+		config config.ShardedNoSQL
+		dc     *p.DynamicConfiguration
+		logger log.Logger
+
+		connectedShards map[string]nosqlStore
+		defaultShard    nosqlStore
+		shardingPolicy  shardingPolicy
+	}
+)
+
+func newShardedNosqlStore(cfg config.ShardedNoSQL, logger log.Logger, dc *p.DynamicConfiguration) (*shardedNosqlStore, error) {
+	sn := shardedNosqlStore{
+		config: cfg,
+		dc:     dc,
+		logger: logger,
+	}
+
+	// Connect to the default shard
+	defaultShardName := cfg.DefaultShard
+	store, err := sn.connectToShard(cfg.Connections[defaultShardName].NoSQLPlugin, defaultShardName)
```

Comment:
it seems that `connectToShard` function only needs one parameter.

Reply:
That's right! I forgot to clean this up while doing refactoring.